aws/cloudformation-templates/segment.yaml

```diff
@@ -0,0 +1,132 @@
+---
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: >
+    This template deploys two Lambda functions that can be used with the Segment Amazon Personalize destination 
+    in the Personalize and CDP workshops, along with the execution roles required to run the Lambda and for Segment
+    to be able to access the Lambda from the Personalize destination.
+
+Parameters:
+  ResourceBucket:
+    Type: String
+    Description: >
+      S3 bucket name where the Retail Demo Store deployment resources are staged (product images, nested CloudFormation templates, source code snapshot, 
+      notebooks, deployment Lambda code, etc).  You can substitute your own bucket here if needed.
+
+  ResourceBucketRelativePath:
+    Type: String
+    Description: >
+      Optional path in the Deployment Resources Staging bucket where the deployment resources are stored (e.g. path/path2/). 
+      Leave blank if resources are at the root of the Staging Resource Bucket. If specified, MUST end with '/'.
+
+Resources:
+  SegmentPersonalizeEventsDestinationLambda:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Description: 'Handles training events passed from Segment to the Personalize tracker for user-item interactions.'
+      Handler: segment-personalize-events-destination.lambda_handler
+      Role: !GetAtt 
+        - SegmentPersonalizeDestinationLambdaExecutionRole
+        - Arn
+      Code:
+        S3Bucket: !Ref ResourceBucket
+        S3Key: !Sub '${ResourceBucketRelativePath}aws-lambda/segment-personalize-events-destination.zip'
+      Runtime: python3.8
+      Timeout: 900
+      FunctionName: SegmentPersonalizeEventsDestination
+      Environment:
+        Variables:
+          personalize_tracking_id:  ''
+
+  SegmentPersonalizeInferenceDestinationLambda:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Description: 'Handles events passed from Segment to Personalize for inference.'
+      Handler: segment-personalize-inference-destination.lambda_handler
+      Role: !GetAtt 
+        - SegmentPersonalizeDestinationLambdaExecutionRole
+        - Arn
+      Code:
+        S3Bucket: !Ref ResourceBucket
+        S3Key: !Sub '${ResourceBucketRelativePath}aws-lambda/segment-personalize-inference-destination.zip'
+      Runtime: python3.8
+      Timeout: 900
+      FunctionName: SegmentPersonalizeInferenceDestination
+      Environment:
+        Variables:
```

Comment:
The variable `recommendations_service_url` looks to be missing. Consider passing the recommendation's service ALB host name for this value. You should be able to pass it through as a template parameter.